GDJS/Runtime/events-tools/objecttools.js

```diff
@@ -316,6 +316,47 @@ gdjs.evtTools.object.pickNearestObject = function(objectsLists, x, y, inverted)
     return true;
 };
 
+gdjs.evtTools.object.raycastObject = function(objectsLists, x, y, angle, dist, varX, varY, inverted) {
+    var matchObject = null;
+    var testSqDist = inverted ? 0 : dist*dist;
+    var resultX = 0;
+    var resultY = 0;
+
+    var lists = gdjs.staticArray(gdjs.evtTools.object.raycastObject);
```

Comment:
Good use of `gdjs.staticArray` 👍

Reply:
I get the `gdjs.staticArray` use, but not sure how everything works exactly, `objectsLists.values(lists)` returns an array with the values of `lists` (the same as `Object.values(lists)`), but the return value is not stored, so no idea what is done here, I leave it there because you use it all the time and it works :/

Reply:
The idea with `gdjs.staticArray` (and _statics) is too avoid allocating objects.
Whenever you're creating an object (`var myObject = {};`) or an array (`var myArray = []`), these objects or array are allocated in memory. When they are not used anymore, they will be garbage collected.

The only problem with garbage collection is that it occurs from time to time (according to memory pressure, the number of objects in memory, etc...) and "pause" the whole Javascript application (i.e: the game) during this time.

For an application, it's not a problem, but for a game it will make a short pause which will be seen as a lag by the player.

By only using `gdjs.staticArray` or `_statics` to store objects and arrays, we make sure that no new temporary object is created - we're only reusing existing objects. This way it should put less pressure on the garbage collector and reduce any lag in particular on mobile devices.

`objectsLists.values(lists)` will modify the array passed as parameter (`lists`), so there is no need to store the result.